crates/orderbook/src/api/post_order.rs

```diff
@@ -262,6 +262,13 @@ impl IntoWarpReply for AddOrderError {
                 super::error("InvalidReplacement", err.to_string()),
                 StatusCode::UNAUTHORIZED,
             ),
+            AddOrderError::MetadataSerializationFailed => reply::with_status(
+                super::error(
+                    "MetadataSerializationFailed",
+                    "quote metadata failed to serialize as json",
+                ),
+                StatusCode::BAD_REQUEST,
```

Comment:
This is more an internal server error, no?
I mean the user can't provide the meta data in their request. The only way this error happens is if we made a mistake.